packages/components/src/components/common/inputbadge/inputBadge.tsx

```diff
@@ -27,23 +27,26 @@ type BadgeProps = {
   tooltipGen?: (text: string) => ReactNode;
 };
 
-const Badge = ({ icon, text, originalText, tooltipGen }: BadgeProps) => (
-  <TooltipProvider>
-    <Tooltip>
-      <TooltipTrigger asChild>
-        <Flex gap={1} alignItems='center' className={inputBadge}>
-          <IvyIcon className={inputBadgeIcon} icon={icon} />
-          {text}
-        </Flex>
-      </TooltipTrigger>
-      {tooltipGen && (
-        <TooltipContent collisionPadding={10} sideOffset={10}>
-          {tooltipGen(originalText)}
-        </TooltipContent>
-      )}
-    </Tooltip>
-  </TooltipProvider>
-);
+const Badge = ({ icon, text, originalText, tooltipGen }: BadgeProps) => {
+  const [tooltipState, setTooltipState] = useState(false);
+  return (
+    <TooltipProvider>
+      <Tooltip onOpenChange={setTooltipState}>
+        <TooltipTrigger asChild>
+          <Flex gap={1} alignItems='center' className={inputBadge}>
+            <IvyIcon className={inputBadgeIcon} icon={icon} />
+            {text}
+          </Flex>
+        </TooltipTrigger>
+        {tooltipState && tooltipGen && (
+          <TooltipContent collisionPadding={10} sideOffset={10}>
+            {tooltipGen(originalText)}
+          </TooltipContent>
+        )}
+      </Tooltip>
+    </TooltipProvider>
+  );
+};
```

Comment:
other which should also work and I personally prefer would be move the call of the function to it's own component. Because the radix-ui tooltip content is only rendered when it will be shown (if it is a react component), if it is like it was before (direct call) then it will be resolved directly. By the way I think it would be good if the original text is show when no tooltipGen function is given. 
```suggestion
const Badge = ({ icon, text, ...props }: BadgeProps) => (
  <TooltipProvider>
    <Tooltip onOpenChange={setTooltipState}>
      <TooltipTrigger asChild>
        <Flex gap={1} alignItems='center' className={inputBadge}>
          <IvyIcon className={inputBadgeIcon} icon={icon} />
          {text}
        </Flex>
      </TooltipTrigger>
      <TooltipContent collisionPadding={10} sideOffset={10}>
        <BadgeTooltip {...props} />
      </TooltipContent>
    </Tooltip>
  </TooltipProvider>
);

const BadgeTooltip = ({originalText, tooltipGen}: Pick<BadgeProps, 'originalText' | 'tooltipGen'>}) => tooltipGen ? tooltipGen(originalText) : originalText;
```

Reply:
:+1: